### DISCUSSION TITLE
How to install and run LR-Free branch?

### DISCUSSION THREAD

**Discussion by runner22k** [bmaltais/kohya_ss#219] [Q&A] [answered]
After watching [recent video](https://youtu.be/etEo6Lf7QIc), I want to try [LR-Free branch](https://github.com/bmaltais/kohya_ss/tree/LR-Free). How to install and run this branch?

**Comment by bmaltais** [comment, marked as answer]
I just published a new version of the GUI that has the dadaptation optimizer built in. You can use it by selecting dadaptation as the optimizer and making sure to set the Learning, Text Encoder Learning rate and Unet learning rate as shown in the image below:

![image](https://user-images.githubusercontent.com/7474674/220803061-791d2432-86e5-4d12-b51d-d4beba06033c.png)

But the best I have found so far is Adafactor... produced the best results for me. Just set the Learning Rate value to 1 and give it a try.

**Reply by runner22k** [reply]
Wow! Thank you so much.

**Reply by runner22k** [reply]
I will copy those values, _Default values in the GUI were these... Learning rate: 0.0001, Text Encoder learning rate: 5e-5, Unet learning rate: 0.0001._ But **We should use Learning rate: 1, Text Encoder learning rate: 0.5 and Unet learning rate: 1**

But Just a request... (I am sorry if I shouldn't be asking this question.)
If Learning, Text Encoder and Unet Learning rate is must, Can it be populated with some default values when user select Optimizer as DAdaptation?

**Reply by bmaltais** [reply]
Hummmm... Technically I could... But some users will complain that switching optimizer mangles the LR values they had manually set... So there is not one solution please all for this... I will see if I can think of something... Like maybe a reset default values button? Those apparently small thing can be daunting to implement because different options will require different defaults... And maintaining this will be a lot of work.

Maybe I will think of something...

**Reply by runner22k** [reply]
:) I agree. default button is also great feature. But you should keep track of all best/workable/accepted values for that. or just add these values in your Help section and add a message/tool tip in the UI.

Where can we find documentation for all the optimizers available in kohya?

**Reply by quasiblob** [reply]
Do we still have to set these default values if Dadaptation is selected?

**Reply by jonwong666** [reply]
@bmaltais could you screen shot the settings to use with adafactor please?

**Comment by bmaltais** [comment]
With adafactor you put 1 for the LR. You can put 0.5 in the Text Encoder to
reduce the learning for it in half.

**Reply by jonwong666** [reply]
so the same as shown above just with adafactor changed in the optimizer? 
![image](https://user-images.githubusercontent.com/41051927/225691221-db036cd8-b58d-4cde-af1f-34fe2efb159e.png)

**Comment by bmaltais** [comment]
Correct
